opm/autodiff/NewtonSolver.hpp

```diff
@@ -66,7 +66,7 @@ namespace Opm {
         /// \param[in]      param   parameters controlling nonlinear Newton process
         /// \param[in, out] model   physical simulation model
         explicit NewtonSolver(const SolverParameters& param,
-                              PhysicalModel& model);
+                              std::shared_ptr<PhysicalModel> model);
```

Comment:
@bska: I don't really understand your question. sorry for my stupidity.
@atgeirr: the shared pointer is necessary to allow the model and the solver to be allocated in a separate method. currently it only worked because the solver and the model live in the same scope (i.e., the `run()` method), thus the model got deallocated after the solver.